src/core/Boxes/boundingbox.cpp

```diff
@@ -865,6 +865,100 @@ void BoundingBox::alignPivot(const Qt::Alignment align, const QRectF& to) {
     alignGeometry(QRectF(pivot, pivot), align, to);
 }
 
+void BoundingBox::alignPivotItself(const Qt::Alignment align,
+                              const QRectF& to,
+                              const AlignRelativeTo relativeTo,
+                              const QPointF lastPivotAbsPos) {
+    auto settings = getStrokeSettings();
+    auto strokeWidth = settings->getCurrentStrokeWidth();
+    
+    auto boundingBox2 = getAbsBoundingRect();
+    auto toAbsBoundingRect = getTotalTransform().mapRect(to);
+    QPointF currentPivot = mTransformAnimator->getPivot();
+    QPointF currentPivotAbsPos = getPivotAbsPos();
+
+    QPointF lastSelectedPivotAbsPos = lastPivotAbsPos;
+    
+    QPointF center = getRelCenterPosition();
+
+    if (relativeTo == AlignRelativeTo::scene) {
+        if (align & Qt::AlignVCenter) {
```

Comment:
We should probably use a switch. Not important, just wanted to note :)

Reply:
OK